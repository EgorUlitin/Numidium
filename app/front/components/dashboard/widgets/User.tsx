import * as React from 'react'

import { Card } from 'antd'

import User from 'model/User'
import Loader from 'components/common/Loader'

import Container from './UserContainer'


export interface Props {
    loading?: boolean
    error?: boolean
    user?: User
}

class UserComponent extends React.PureComponent<Props, {}> {
    render() {
        const { user, loading, error } = this.props

        return (
            <React.Fragment>
                <Card cover={
                    <img alt='avatar' src='https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png' />
                }>
                    <Loader loading={loading || !user} error={error}>
                        {!!user && <Card.Meta title={user && user.email} />}
                    </Loader>
                </Card>
            </React.Fragment>
        )
    }
}

export default Container(UserComponent)
