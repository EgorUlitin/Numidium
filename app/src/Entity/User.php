<?php

namespace App\Entity;


use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Security\Core\User\UserInterface;

/**
 * @ORM\Table(name="app_users")
 * @ORM\Entity()
 */
class User implements UserInterface, \Serializable
{
    /**
     * @ORM\Column(type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /** @ORM\Column(type="string", length=127) */
    private $password;

    /** @ORM\Column(type="string", length=127, unique=true) */
    private $email;

    /** @ORM\Column(type="boolean") */
    private $active = false;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUsername(): ?string
    {
        return $this->email;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): User
    {
        $this->email = $email;

        return $this;
    }

    public function getSalt(): ?string
    {
        return null;
    }

    public function getPassword(): ?string
    {
        return $this->password;
    }

    public function setPassword(string $password): User
    {
        $this->password = $password;

        return $this;
    }

    public function isActive(): bool
    {
        return $this->active;
    }

    public function setActive(bool $flag): User
    {
        $this->active = $flag;

        return $this;
    }

    public function getRoles(): array
    {
        return [
            'ROLE_USER',
        ];
    }

    public function eraseCredentials(): void
    {

    }

    /** @see \Serializable::serialize() */
    public function serialize(): string
    {
        return serialize([
            $this->id,
            $this->email,
            $this->password,
        ]);
    }

    /** @see \Serializable::unserialize() */
    public function unserialize($serialized): void
    {
        list (
            $this->id,
            $this->email,
            $this->password,
            ) = unserialize($serialized);
    }
}