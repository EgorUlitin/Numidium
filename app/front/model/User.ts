import Entity from './Entity'

export interface Owner extends Entity {
    id: number
}

export default interface User extends Entity {
    id: number
    email: string
    active: boolean
}
