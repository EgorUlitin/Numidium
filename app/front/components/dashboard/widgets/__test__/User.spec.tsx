import * as React from 'react'
import * as renderer from 'react-test-renderer'
import User from 'model/User'

import { UserComponent } from '../User'


test('`User` widget renders correctly', () => {
    const user = {
        '@id': '1',
        '@type': 'TestUser',
        id: 1,
        email: 'test@example.com',
        active: true,
    } as User

    const tree = renderer
        .create(<UserComponent user={user} />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})
