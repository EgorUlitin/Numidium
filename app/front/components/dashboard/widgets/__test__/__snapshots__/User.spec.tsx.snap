// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`User\` widget renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="avatar"
      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
    />
  </div>
  <div
    className="ant-card-body"
    style={undefined}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-title"
        >
          test@example.com
        </div>
      </div>
    </div>
  </div>
</div>
`;
